src/reporter.ts

```diff
@@ -34,21 +35,34 @@ export class Reporter {
   private launchId: string;
   private suiteIds: string[];
   private testItems: TestItem[];
-  private testData: { [name: string]: string };
+  private testData: ObjUniversal;
+  private customLaunchStatus: string;
 
   constructor(config: ReportPortalConfig) {
     this.noColors = false;
     this.suiteIds = [];
     this.testItems = [];
     this.testData = {};
+    this.customLaunchStatus = '';
 
     const agentInfo = getAgentInfo();
 
     this.config = config;
     this.client = new RPClient(config, agentInfo);
   }
 
-  reportTaskStart(startTime: number, userAgents: any, testCount: number): void {
+  registerRPListeners(): void {
+    process.on(EVENTS.SET_LAUNCH_STATUS, this.setLaunchStatus.bind(this));
+    process.on(EVENTS.SET_STATUS, this.setStatus.bind(this));
+  }
+
+  unregisterRPListeners(): void {
+    process.removeAllListeners(EVENTS.SET_LAUNCH_STATUS);
+    process.removeAllListeners(EVENTS.SET_STATUS);
```

Comment:
Please use [process.off](https://nodejs.org/api/events.html#events_emitter_off_eventname_listener) to remove necessary listeners to be more consistent.